extensions/common/vault/vault-hashicorp/src/main/java/org/eclipse/edc/vault/hashicorp/HashicorpVaultClient.java

```diff
@@ -24,76 +25,206 @@
+            }
+        } catch (IOException e) {
+            var errMsg = "Failed to look up token with reason: %s".formatted(e.getMessage());
+            monitor.warning(errMsg, e);
+            return Result.failure(errMsg);
+        }
+    }
+
+    /**
+     * Attempts to renew the Vault token with the configured ttl. Note that Vault will not honor the passed
+     * ttl (or increment) for periodic tokens. Therefore, the ttl returned by this operation should always be used
+     * for further calculations.
+     * <p>
+     * Will retry in some error cases.
+     *
+     * @return long representing the remaining ttl of the token in seconds
+     */
+    public Result<Long> renewToken() {
+        var uri = configValues.url()
+                .newBuilder()
+                .addPathSegments(TOKEN_RENEW_SELF_PATH)
+                .build();
+        var requestPayload = TokenRenewRequest.Builder
+                .newInstance()
+                .increment(configValues.ttl())
+                .build();
+        var request = httpPost(uri, requestPayload);
+
+        try (var response = httpClient.execute(request, FALLBACK_FACTORIES)) {
+            if (response.isSuccessful()) {
+                var responseBody = response.body();
+                if (responseBody == null) {
+                    return Result.failure("Token renew returned empty body");
+                }
+                var payload = objectMapper.readValue(responseBody.string(), MAP_TYPE_REFERENCE);
+                var parseTtlResult = parseTtl(payload);
+                if (parseTtlResult.failed()) {
+                    return Result.failure("Token renew response could not be parsed: %s".formatted(parseTtlResult.getFailureDetail()));
+                }
+                var ttl = parseTtlResult.getContent();
+                return Result.success(ttl);
+            } else {
+                return Result.failure("Token renew failed with status: %d".formatted(response.code()));
+            }
+        } catch (IOException e) {
+            var errMsg = "Failed to renew token with reason: %s".formatted(e.getMessage());
+            monitor.warning(errMsg, e);
+            return Result.failure(errMsg);
+        }
     }
 
     public Result<String> getSecretValue(@NotNull String key) {
         var requestUri = getSecretUrl(key, VAULT_SECRET_DATA_PATH);
-        var headers = getHeaders();
-        var request = new Request.Builder().url(requestUri).headers(headers).get().build();
+        var request = httpGet(requestUri);
 
         try (var response = httpClient.execute(request)) {
 
             if (response.isSuccessful()) {
                 if (response.code() == HTTP_CODE_404) {
-                    return Result.failure(
-                            String.format(CALL_UNSUCCESSFUL_ERROR_TEMPLATE, "Secret not found"));
+                    return Result.failure("Secret %s not found".formatted(key));
```

Comment:
The secret key should not be recorded. The client knows which key it asked for and can handle it appropriately.